### PR TITLE
Build for Apple Silicon

### LINKED ISSUE #171: Release binaries for Apple Silicon
Seems go is not going to support M1 until 1.16 but it [seems it's possible to cross compile arm binary](https://github.com/golang/go/issues/38485#issuecomment-729296273). We should try to build the CLI and the agent binary this way.

### PR DESCRIPTION
**Pull request by fkorotkov**
Via gotip

Will test binary on Friday when the new Mac Mini should arrive.

Related issues: https://github.com/golang/go/issues/42684

Fixes #171

### CODE REVIEW

**Comment by edigaryev** [comment]
These two tasks look very similar, can we use [YAML aliases](https://cirrus-ci.org/guide/tips-and-tricks/#sharing-configuration-between-tasks) instead to reduce the duplication?

**Comment by fkorotkov** [comment]
Though about it but since these tasks are temporary until Goreleaser support Apple Silicon releases I decided not to optimize.  There are slightly different build scripts so only `env`, `install_script` and `build_script` can be refactored in an alias.

File: .cirrus.yml [hunk 1]
```diff
@@ -41,21 +41,74 @@ docker_builder:
...
     image: goreleaser/goreleaser:latest
   release_script: goreleaser build --snapshot
   binaries_artifacts:
     path: "dist/cirrus_*/cirrus*"
 
+task:
+  name: Release Apple Silicon (Dry Run)
+  only_if: $CIRRUS_TAG == ''
+  env:
+    GITHUB_TOKEN: ENCRYPTED[!98ace8259c6024da912c14d5a3c5c6aac186890a8d4819fad78f3e0c41a4e0cd3a2537dd6e91493952fb056fa434be7c!]
```
Comment (edigaryev) [inline .cirrus.yml:55]: This doesn't seem to be required for the dry-run task.

**Comment by edigaryev** [comment]
> There are slightly different build scripts

Why not make them similar? It seems that the only difference should be that we don't upload the results.

File: .cirrus.yml (hunk 1 above)
Comment (fkorotkov) [reply .cirrus.yml]: Fixed in b8db09f6caab7badf9989d71c4caff2cfe66f750

**Comment by fkorotkov** [comment]
Refactored out the parts in b8db09f6caab7badf9989d71c4caff2cfe66f750. Also bad news, Mac Mini will arrive only on Tuesday. 😪

**Review by edigaryev — APPROVED** [review]

**Comment by fkorotkov** [comment]
It works! 
image

**Pull request merged**